client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -151,7 +150,11 @@ const Checkout = React.createClass( {
 			return false;
 		}
 
-		if ( this.props.transaction.step.name === transactionStepTypes.SUBMITTING_WPCOM_REQUEST ) {
+		// Do not redirect if cart is empty, but we're submitting or received transaction response
+		if ( includes(
+			[ transactionStepTypes.SUBMITTING_WPCOM_REQUEST, transactionStepTypes.RECEIVED_WPCOM_RESPONSE ],
```

Comment:
If you empty the cart at that point, you're aren't redirected out of checkout:

![screen shot 2017-02-13 at 9 51 20 am](https://cloud.githubusercontent.com/assets/1130674/22895636/104d4150-f1d2-11e6-9fef-3a9b2c40e6f6.png)